github-cli.md

```diff
@@ -0,0 +1,110 @@
+
+**Windows: ** `cd` stands for "change directory" and can be used to switch to a new directory as well
+
+#### Learn More About a Command
+**OS X and Linux: ** `man <command>`
+
+**Windows: ** `<command> /?`
+
+#### List Files and Directories
+**OS X and Linux: ** `ls`
+
+**Windows: ** `dir`
+
+#### Change Current Directory
+**OS X, Linux, and Windows: ** `cd <path>`
+
+#### Move to the Parent Directory
+**OS X, Linux, and Windows: ** `cd ..`
+
+#### Create Directory
+**OS X, Linux, and Windows: ** `mkdir <name of directory>`
+
+### Cheat Sheet
+| Command | OS X and Linux | Windows |
+| :-----: | :------------: | :-----: |
+| Current Directory | `pwd` |  `cd`  |
+| Learn More | `man <command>` | `<command> /?` |
+| List Files and Directories | `ls` | `dir` |
+| Change Current Directory | `cd <path>` | `cd <path` |
+| Move to Parent Directory | `cd ..` | `cd ..` |
+| Create Directory | `mkdir <name of directory`> | `mkdir <name of directory>` |
+
+
+## **Github**
+<img src="https://memegenerator.net/img/instances/41740049/regular-peopleadd-me-on-facebook-me-fork-me-on-github.jpg" width="300px" height="300px"/>
+
+### Introduction
+**What: ** Git is a version control system which allows multiple people to work on a project at the same time (kind of like Google Docs).
+
+**Why: ** Imagine cooking with your friends: you all have to keep track of what the others are adding to the dish while making sure that what you add doesn't ruin it. Things get more complicated as more and more people join. The same is true of technical projects.
+
+**How: ** [Github!](https://github.com/)
+
+### Commands
+#### `git init <repository name`
+This will create a new git repository on your **local** machine
+
+#### `git clone <link to Github repository>`
+This allows you to create a copy of the project on your local machine. When you make changes, they'll only show up on your copy, we'll talk about how to **push** your changes to the remote repository in another command
+
+#### `git checkout [-b <new branch name> | <existing branch>]`
+<img src="https://wac-cdn.atlassian.com/dam/jcr:86eba9ec-9391-45ea-800a-948cec1f2ed7/Branch-2.png?cdnVersion=lf" width="400px" height="300px"/>
+
+Branching is useful when you want to add a feature to the project without affecting the code of others working on it. It allows you to work on your feature without worrying about what others are working on *for now*
+
+#### `git add [<file/directory> | .]`
+Adding a file means that you want to move a file from the working directory to the staging directory. These files will be added to the next **commit** which will be explained in the next command. **Note: ** `git add .` indicates that you want to add all the files you just updated to the staging directory.
+
+#### `git commit -m "<commit message>"`
+This command is used to save the changes you made in your local repository. The commit message is often a sentence or two explaining what you changed and why.
```

Comment:
Typically commits are stored with reference codes that are not human-readable. Commit messages help differentiate versions of code at different points in time.